wire-ios-request-strategy/Sources/E2EIdentity/EnrollE2eICertificateUseCase.swift

```diff
@@ -33,37 +39,40 @@ public final class EnrollE2eICertificateUseCase: EnrollE2eICertificateUseCaseInt
         self.e2eiRepository = e2eiRepository
     }
 
-    public func invoke(idToken: String, selfUser: ZMUser) async throws -> String {
-        guard
-            let userName = selfUser.name,
-            let handle = selfUser.handle,
-            let e2eiClientId = E2eIClientID(user: selfUser)
-        else {
-            throw EnrollE2EICertificateUseCaseFailure.failedToSetupEnrollment
-        }
-
-        let enrollment = try await e2eiRepository.createEnrollment(e2eiClientId: e2eiClientId, userName: userName, handle: handle)
+    public func invoke(e2eiClientId: E2eIClientID,
+                       userName: String,
+                       userHandle: String,
+                       invokeAuthentication: OAuthBlock) async throws -> String {
```

Comment:
why not use a nice verb instead

```suggestion
                       authenticate: OAuthBlock) async throws -> String {
```